src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -130,22 +116,30 @@ public ElasticsearchClient(
                     .toArray(new HttpHost[config.connectionUrls().size()])
             )
             .setHttpClientConfigCallback(configCallbackHandler)
-            .setRequestConfigCallback(configCallbackHandler)
     );
     this.bulkProcessor = BulkProcessor
-        .builder((req, lis) -> client.bulkAsync(req, RequestOptions.DEFAULT, lis), buildListener())
+        .builder(buildConsumer(), buildListener())
         .setBulkActions(config.batchSize())
         .setConcurrentRequests(config.maxInFlightRequests() - 1) // 0 = no concurrent requests
         .setFlushInterval(TimeValue.timeValueMillis(config.lingerMs()))
-        .setBackoffPolicy(
```

Comment:
is this removed because we dont want bulk processor to retry at all (even the non IO exceptions?)

Reply:
like what if ES is temporarily sending 429 and the bulk processor was retrying after backingoff, will afterbulk be called in that case?

Reply:
Correct, our retry mechanism is a superset of what BulkProcessor does. 
Added a test.

Reply:
Got it but is there any downside to not using the built in retry?

Reply:
I can't think of any.
The clear downside is to have retry logic split in two different places.